src/compiler/commandLineParser.ts

```diff
@@ -1843,7 +1843,9 @@ namespace ts {
             if (hasProperty(raw, "files") && !isNullOrUndefined(raw.files)) {
                 if (isArray(raw.files)) {
                     filesSpecs = <ReadonlyArray<string>>raw.files;
-                    if (filesSpecs.length === 0) {
+                    const hasReferences = hasProperty(raw, "references") && !isNullOrUndefined(raw.references);
+                    const hasZeroOrNoReferences = !hasReferences || raw.references.length === 0;
+                    if (filesSpecs.length === 0 && hasZeroOrNoReferences) {
                         createCompilerDiagnosticOnlyIfJson(Diagnostics.The_files_list_in_config_file_0_is_empty, configFileName || "tsconfig.json");

```

Comment:
This will not report error correctly in tsconfig file. Since this error will only be reported if tsconfig was parsed with text of json file instead of creating TsConfigSourceFile

Reply:
You also want to report this error correctly on files node when TsConfigSourceFile is present